test/BreakInvariantBounty.test.js

```diff
@@ -17,82 +17,95 @@ const sendReward = async (from, to, value) => ethSendTransaction({
-      const claim = await this.bounty.claimed();
-
-      claim.should.equal(true);
-
-      const researcherPrevBalance = await ethGetBalance(researcher);
+    describe('destroy', function () {
+      it('returns all balance to the owner', async function () {
+        const ownerPreBalance = await ethGetBalance(owner);
+        await this.bounty.destroy({ from: owner, gasPrice: 0 });
+        const ownerPostBalance = await ethGetBalance(owner);
 
-      await this.bounty.withdrawPayments(researcher, { gasPrice: 0 });
-      const updatedBalance = await ethGetBalance(this.bounty.address);
-      updatedBalance.should.be.bignumber.equal(0);
+        (await ethGetBalance(this.bounty.address)).should.be.bignumber.equal(0);
+        ownerPostBalance.sub(ownerPreBalance).should.be.bignumber.equal(reward);
+      });
 
-      const researcherCurrBalance = await ethGetBalance(researcher);
-      researcherCurrBalance.sub(researcherPrevBalance).should.be.bignumber.equal(reward);
+      it('reverts when called by anyone', async function () {
+        await assertRevert(this.bounty.destroy({ from: anyone }));
+      });
     });
 
-    it('cannot claim reward from non-target', async function () {
-      await assertRevert(
-        this.bounty.claim(nonTarget, { from: researcher })
-      );
-    });
+    describe('claim', function () {
+      it('is initially unclaimed', async function () {
+        (await this.bounty.claimed()).should.equal(false);
+      });
+
+      it('can create claimable target', async function () {
+        const { logs } = await this.bounty.createTarget({ from: researcher });
+        expectEvent.inLogs(logs, 'TargetCreated');
+      });
 
-    context('reward claimed', function () {
-      beforeEach(async function () {
-        await this.bounty.claim(this.targetAddress, { from: researcher });
+      context('with target', async function () {
+        beforeEach(async function () {
+          const { logs } = await this.bounty.createTarget({ from: researcher });
+          const event = expectEvent.inLogs(logs, 'TargetCreated');
+          this.target = TargetMock.at(event.args.createdAddress);
+        });
+
+        context('before exploiting vulnerability', async function () {
+          it('reverts when claiming reward', async function () {
+            await assertRevert(this.bounty.claim(this.target.address, { from: researcher }));
+          });
+        });
+
+        context('after exploiting vulnerability', async function () {
+          beforeEach(async function () {
+            await this.target.exploitVulnerability({ from: researcher });
+          });
+
+          it('sends the reward to the researcher', async function () {
+            await this.bounty.claim(this.target.address, { from: anyone });
+
+            const researcherPreBalance = await ethGetBalance(researcher);
+            await this.bounty.withdrawPayments(researcher);
+            const researcherPostBalance = await ethGetBalance(researcher);
```

Comment:
It would be nice to abstract this pattern into a test helper.

Reply:
Created https://github.com/OpenZeppelin/openzeppelin-solidity/issues/1355